### PR TITLE
Kelsonic 28848 contact us widget

### PR DESCRIPTION
**Pull request by kelsonic**
## Description

This PR implements the `ask-va` react widget for content editors. This PR isn't testable until a react widget content block is added in drupal with the react widget ID `ask-va`.

## Original issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/28848

## Testing done
Added unit tests.

## Acceptance criteria
- [x] Add `ask-va` react widget

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/static-pages/static-pages-entry.js [hunk 1]
```diff
@@ -1,93 +1,79 @@
...
 import LazyLoad from 'vanilla-lazyload/dist/lazyload';
 import * as Sentry from '@sentry/browser';
-
-import createCommonStore from 'platform/startup/store';
-import startSitewideComponents from 'platform/site-wide';
-import { VA_FORM_IDS } from 'platform/forms/constants';
-
+// Relative imports.
 import './analytics';
+import './sass/static-pages.scss';
```
Comment (kelsonic) [inline src/applications/static-pages/static-pages-entry.js:6]: This is all just reorganizing to make it more legible 🙂 (except the ask-va part)

File: src/applications/static-pages/widgetTypes.js [hunk 2]
```diff
@@ -1,13 +1,16 @@
 export default {
+  ASK_VA: 'ask-va',
```
Comment (kelsonic) [inline src/applications/static-pages/widgetTypes.js:2]: This is new

File: src/applications/static-pages/ask-va/helpers/index.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+// Relative imports.
+import environment from 'platform/utilities/environment';
+
+export const deriveDefaultURL = () => {
+  // Production environments.
+  if (environment.isProduction()) {
+    return 'https://ava.va.gov/';
```
Comment (zacharymorel) [inline src/applications/static-pages/ask-va/helpers/index.js:7]: So this URL is not live yet?

**Review by zacharymorel — APPROVED** [review]

File: src/applications/static-pages/ask-va/components/App/index.unit.spec.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import { App } from '.';
+
+describe('Ask VA <App>', () => {
+  it('does not render when explicitly hidden', () => {
+    const wrapper = shallow(<App hidden />);
+    expect(wrapper.text()).to.equal('');
+    wrapper.unmount();
+  });
+
+  it('renders what we expect when not authenticated', () => {
```
Comment (Rleahy22) [inline src/applications/static-pages/ask-va/components/App/index.unit.spec.js:16]:
```suggestion
  it('renders ask va link when not authenticated', () => {
```

File: src/applications/static-pages/ask-va/components/App/index.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+import { connect } from 'react-redux';
+// Relative imports.
+import {
+  deriveLOA1URL,
+  deriveLOA2PlusURL,
+  deriveDefaultURL,
+} from '../../helpers';
+import { selectProfile } from 'platform/user/selectors';
+
+export const App = ({ loa, hidden }) => {
```
Comment (zacharymorel) [inline src/applications/static-pages/ask-va/components/App/index.js:13]: I enjoy the simplicity of the app and utilizing the helpers in an adjacent file.

File: src/applications/static-pages/ask-va/helpers/index.js (hunk 3 above)
Comment (kelsonic) [reply src/applications/static-pages/ask-va/helpers/index.js:7]: That's correct, yupp! Not until October I think?

File: src/applications/static-pages/ask-va/components/App/index.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  // Do not render if the widget is hidden.
+  if (hidden) {
+    return null;
```
Comment (kelsonic) [inline src/applications/static-pages/ask-va/components/App/index.js:16]: We are currently hiding the widget on production environments.

File: src/applications/static-pages/ask-va/components/App/index.unit.spec.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    expect(wrapper.find(`a[href="${expectedHref}"]`)).be.have.lengthOf(1);
+    expect(wrapper.find(`a[href="${expectedHref}"]`).text()).to.equal(
+      'Ask us a question online',
+    );
+    wrapper.unmount();
+  });
+
+  it('renders what we expect when user is LOA1', () => {
+    const expectedHref = environment.isProduction()
+      ? 'https://ava.va.gov/Unauthenticated'
```
Comment (Rleahy22) [inline src/applications/static-pages/ask-va/components/App/index.unit.spec.js:30]: If they're authenticated we direct them to an unauthenticated page?

File: src/applications/static-pages/ask-va/helpers/index.js (hunk 3 above)
Comment (zacharymorel) [reply src/applications/static-pages/ask-va/helpers/index.js:7]: DERP, I just saw your comment below, disregard.

**Review by Rleahy22 — APPROVED** [review]

File: src/applications/static-pages/ask-va/helpers/index.js (hunk 3 above)
Comment (kelsonic) [reply src/applications/static-pages/ask-va/helpers/index.js:7]: haha yupp the react widget is hidden on production currently! 🙂

File: src/applications/static-pages/ask-va/components/App/index.unit.spec.js (hunk 7 above)
Comment (kelsonic) [reply src/applications/static-pages/ask-va/components/App/index.unit.spec.js:30]: If they are LOA1, correct 🙂

Comment (kelsonic) [reply src/applications/static-pages/ask-va/components/App/index.unit.spec.js:30]: For context: https://github.com/department-of-veterans-affairs/va.gov-team/issues/28848#issuecomment-907259727

Comment (kelsonic) [reply src/applications/static-pages/ask-va/components/App/index.unit.spec.js:30]: image

Comment (kelsonic) [reply src/applications/static-pages/ask-va/components/App/index.unit.spec.js:30]: I'm assuming it's this way due to LOA1 users not having necessary credentials (first name, last name, email, etc.) for a personalized ask VA experience, but just a guess 🙃

**Pull request merged**